src/main/scala/mesosphere/marathon/storage/migration/MigrationTo17.scala

```diff
@@ -99,8 +99,11 @@ object MigrationTo17 extends StrictLogging {
       (__ \ "tasksMap").read[Map[Task.Id, Task]](taskMapReads17) ~
       (__ \ "runSpecVersion").read[Timestamp] ~
       (__ \ "state").read[InstanceState](instanceStateReads160) ~
-      (__ \ "reservation").readNullable[Reservation]
-    ) { (instanceId, agentInfo, tasksMap, runSpecVersion, state, reservation) =>
+      (__ \ "reservation").readNullable[JsObject]
+    ) { (instanceId, agentInfo, tasksMap, runSpecVersion, state, rawReservation) =>
+        val reservation = rawReservation.map { raw =>
+          raw.as[Reservation](InstanceMigration.legacyReservationReads(tasksMap, instanceId))
```

Comment:
Do you think we should not depend on the domain logic in the migration?

Reply:
I don't see another way around this. All migrations have to migrate the reservation id as well. I'm happy to see an alternative.

Reply:
I think it's fine, we do have a unit test, right?